parquet/src/main/scala/magnolify/parquet/Schema.scala

```diff
@@ -77,19 +84,28 @@ private object Schema {
     builder.named(schema.getName)
   }
 
-  def pruneRequested(fileSchema: MessageType, requestedSchema: MessageType): MessageType = {
-    def prune(file: GroupType, requested: GroupType): Type = {
-      val filtered = requested.getFields.asScala
-        .flatMap { rf =>
-          if (file.containsField(rf.getName)) {
```

Comment:
is there any regression now that we're no longer pruning schema fields that don't exist in `fileSchema` ? do we need to add that back in?

Reply:
that's a good question. Based on my observations pruning of a reader schema isn't required i.e. it is ok to pass new optional fields to the `ParquetReader` which don't exist in the writer schema. I checked this specifically in my sandbox environment and the whole schema evolution and projection suits aren't indicating any issues with removing the pruning. And it makes sense in general Parquet lib supports non-existing nullable fields in general, they wouldn't be involved into the decoding process at all i.e. converters wouldn't be created for them.

So basically we are replacing pruning with the `checkCompatibility` function.